controllers/profiles/common/properties/discovery.go

```diff
@@ -99,6 +99,7 @@ func generateDiscoveryProperties(ctx context.Context, catalog discovery.ServiceC
 					mpProperty := generateMicroprofileServiceCatalogProperty(plainUri)
 					klog.V(log.I).Infof("Generating microprofile service catalog property %s=%s.", mpProperty, address)
 					result.MustSet(mpProperty, address)
+					result.MustSet(k, "http://localhost:8080")
```

Comment:
Are we adding `localhost` to discovered endpoints? That way we may lose track of it and we can't refresh the discovery if the service/endpoint changes.

Reply:
this is a workaround to avoid the problem we saw yesterday.
if the `managed-props` does not override the value of the original property, the platform would anyway parse it and may raise a startup error like:
```
2024-01-29 16:15:52,397 orderprocessing-5f658b4c9d-wtmnp ERROR [io.quarkus.runtime.Application:-1] (Quarkus Main Thread) Failed to start application (with profile [dev]): java.lang.RuntimeException: Failed to start quarkus
...
Caused by: jakarta.enterprise.inject.spi.DeploymentException: java.lang.IllegalArgumentException: Unsupported scheme for the http connector in URL: services.v1/event-listener
...
```
(the property value being `${kubernetes:services.v1/event-listener}`)

If there is a more elegant workaround, we can give it a try, but for sure we cannot remove the original property from the user configuration as we did before.

Reply:
@wmedvede we still have to assess this problem on the runtimes side. When we have a profile and standard properties in a discovery context.

Reply:
I was wondering how I could create such discovery context, in fact. Our workflows are missing the `kogito-addons-quarkus-fabric8-kubernetes-service-catalog` extension, should we add it?
I've updated one of our application to define an openAPI endpoint as:
```
mta_ui = ${kubernetes:services.v1/openshift-mta/mta-ui?port=ui}
quarkus.rest-client.mta_json.url = ${mta_ui}/hub
```
which is then mapped into the managed properties:
```
mta_ui = http://localhost:8080
org.kie.kogito.addons.discovery.kubernetes\:services.v1\/openshift-mta\/mta-ui?port\=ui = http://mta-ui.openshift-mta.svc:8080
```
But this I believe needs that particular extension to be resolved, correct?

Reply:
I think I solved my specific use case by changing to:
```
result.MustSet(k, address)
```
that leads to these two managed properties instead:
```
org.kie.kogito.addons.discovery.kubernetes\:services.v1\/openshift-mta\/mta-ui?port\=ui = http://mta-ui.openshift-mta.svc:8080
mta_ui = http://mta-ui.openshift-mta.svc:8080
```
where the first one is not in use because of the missed extension (my guess)

Reply:
@dmartinol we ditched the `addons-quarkus-fabric8-kubernetes-service-catalog` on purpose since we implemented the discovery in the operator, to not depend on Fabric8 and giving the pod unneeded permissions.

That said, the underlying runtime now relies on smallrye mpconfig to do the "discovery", which is this props dictionary we mount in the CM. So now we use the [microprofile-config catalog](https://github.com/apache/incubator-kie-kogito-runtimes/tree/main/quarkus/addons/microprofile-config-service-catalog) instead.

So in your example, our managed props take over `mta_ui`? Since we will have:

User Properties:

```properties
mta_ui = ${kubernetes:services.v1/openshift-mta/mta-ui?port=ui}
quarkus.rest-client.mta_json.url = ${mta_ui}/hub
```
Managed:
```
org.kie.kogito.addons.discovery.kubernetes\:services.v1\/openshift-mta\/mta-ui?port\=ui = http://mta-ui.openshift-mta.svc:8080
mta_ui = http://mta-ui.openshift-mta.svc:8080
```

Hence, `mta_ui` must override what's set on the user profile. It will probably work since the profile takes precedence, right?

That will work considering we keep updating `mta_ui` on every reconciliation if a user changes their service endpoint.

Reply:
@ricardozanini I finally understood that `we implemented the discovery in the operator`, as you said. After reading the nice introduction to the [service discovery add-on](https://blog.kie.org/2022/10/kogito-service-discovery-for-workflows.html) by Filippe, I believe this is the way to go.

I will raise an updated commit after a quick validation.

Reply:
Thank you!

Reply:
Latest commits include the above change and I manually validated it on an OCP deployment with the above use case.